### PR TITLE
Change internal names

### PR DESCRIPTION
**Pull request by bsudekum**
This changes internal names to make the UI the priority while all non UI logic is now renamed to core.

* Module name:
  * ~~`MapboxNavigation`~~ `MapboxCoreNavigation`
  * ~~`MapboxNavigationUI`~~ `MapboxNavigation`
* Pod name:
  * ~~`MapboxNavigation`~~ `MapboxCoreNavigation`
  * ~~`MapboxNavigationUI`~~ `MapboxNavigation`

/cc @1ec5 @ericrwolfe

### CODE REVIEW

File: MapboxCoreNavigation/DistanceFormatter.swift [hunk 1]
```diff
@@ -6,32 +7,32 @@ let yardsPerMile = 1_760.0
 let feetPerMile = yardsPerMile * 3.0
 
 /// Provides appropriately formatted, localized descriptions of linear distances.
-class DistanceFormatter: LengthFormatter {
+public class DistanceFormatter: LengthFormatter {
     /// True to favor brevity over precision.
-    var approximate: Bool
+    var approx: Bool
```
Comment (1ec5) [inline MapboxCoreNavigation/DistanceFormatter.swift]: What does this change have to do with renaming the repositories? Can it be split into a separate PR?

(Also, I like the irony of abbreviating a property described as “favoring brevity over precision”. 😄)

File: MapboxCoreNavigation/MapboxCoreNavigation.h [hunk 2]
```diff
@@ -0,0 +1,9 @@
+#import <Foundation/Foundation.h>
+
+//! Project version number for MapboxNavigation.
```
Comment (1ec5) [inline MapboxCoreNavigation/MapboxCoreNavigation.h]: These comments are outdated. (×4)

**Comment by 1ec5** [comment]
Tests are failing:

```
▸ Processing Info.plist

❌  error: no such file or directory: /Users/vagrant/git/MapboxNavigationTests/MapboxNavigationTests.swift



❌  error: no such file or directory: /Users/vagrant/git/MapboxNavigationTests/GeometryTests.swift



❌  error: no such file or directory: /Users/vagrant/git/MapboxNavigationTests/RouteProgressTests.swift



❌  error: no such file or directory: /Users/vagrant/git/MapboxNavigationTests/Fixture.swift
```

File: MapboxNavigation.podspec [hunk 3]
```diff
@@ -2,12 +2,12 @@ Pod::Spec.new do |s|
...
 
-  s.name = "MapboxNavigationUI.swift"
-  s.version = "0.1.0"
+  s.name = "MapboxNavigation"
+  s.version = "0.2.0"
   s.summary = "Mapbox Navigation UI library"
 
   s.description  = <<-DESC
-  MapboxNavigationUI.swift makes it easy to get step by step UI for guiding a user along a route.
+  MapboxNavigation makes it easy to get step by step UI for guiding a user along a route.
```
Comment (1ec5) [inline MapboxNavigation.podspec]: Let’s refer to this library as “Mapbox Navigation” or “The Mapbox Navigation SDK” in prose.

File: MapboxNavigation.xcodeproj/project.pbxproj [hunk 4]
```diff
@@ -291,34 +289,34 @@
 		354BD4F61E6638E800AE1344 /* AWSCore.framework */ = {isa = PBXFileReference; lastKnownFileType = wrapper.framework; name = AWSCore.framework; path = Carthage/Build/iOS/AWSCore.framework; sourceTree = "<group>"; };
 		358D14631E5E3B7700ADE590 /* Example-Swift.app */ = {isa = PBXFileReference; explicitFileType = wrapper.application; includeInIndex = 0; path = "Example-Swift.app"; sourceTree = BUILT_PRODUCTS_DIR; };
 		358D14651E5E3B7700ADE590 /* AppDelegate.swift */ = {isa = PBXFileReference; lastKnownFileType = sourcecode.swift; path = AppDelegate.swift; sourceTree = "<group>"; };
-		358D14671E5E3B7700ADE590 /* ViewController.swift */ = {isa = PBXFileReference; lastKnownFileType = sourcecode.swift; path = ViewController.swift; sourceTree = "<group>"; };
+		358D14671E5E3B7700ADE590 /* ViewController.swift */ = {isa = PBXFileReference; lastKnownFileType = sourcecode.swift; lineEnding = 0; path = ViewController.swift; sourceTree = "<group>"; xcLanguageSpecificationIdentifier = xcode.lang.swift; };
```
Comment (1ec5) [inline MapboxNavigation.xcodeproj/project.pbxproj]: Weird that the line ending changed. Can you double-check that we didn’t wind up with Windows line endings? 😛

File: MapboxNavigation.xcodeproj/project.pbxproj [hunk 5]
```diff
@@ -631,6 +628,7 @@
 				C52AC1251DF0E48600396B9F /* RouteProgressTests.swift */,
 				C5ADFBD91DDCC7840011824B /* Info.plist */,
 			);
+			name = MapboxCoreNavigationTests;
 			path = MapboxNavigationTests;
```
Comment (1ec5) [inline MapboxNavigation.xcodeproj/project.pbxproj]: This is probably the cause of the test failures: the paths associated with the test bundles’ groups are out of date.

File: MapboxNavigation/DistanceFormatter.swift [hunk 6]
```diff
@@ -7,32 +6,32 @@ let yardsPerMile = 1_760.0
 let feetPerMile = yardsPerMile * 3.0
 
 /// Provides appropriately formatted, localized descriptions of linear distances.
-public class DistanceFormatter: LengthFormatter {
+class DistanceFormatter: LengthFormatter {
```
Comment (1ec5) [inline MapboxNavigation/DistanceFormatter.swift]: Why are there separate DistanceFormatter classes in MapboxCoreNavigation and MapboxNavigation? Let’s delete this one and have MapboxNavigation use the one out of MapboxCoreNavigation.

**Comment by bsudekum** [comment]
Low alert test is failing... really not sure why.

**Comment by bsudekum** [comment]
Unsure about this error but there still seems to be some leftover `MapboxNavigationUI`. Can't find it within xcode, only a text editor.

File: MapboxNavigation.xcodeproj/project.pbxproj [hunk 7]
```diff
@@ -674,9 +667,9 @@
 			dependencies = (
 				35CEA3581E5CEBBC009F2255 /* PBXTargetDependency */,
 			);
-			name = MapboxNavigationUI;
+			name = MapboxNavigation;
 			productName = MapboxNavigationUI;
```
Comment (1ec5) [inline MapboxNavigation.xcodeproj/project.pbxproj]: Try changing this to MapboxNavigation using a text editor.

File: MapboxNavigation.xcodeproj/project.pbxproj [hunk 8]
```diff
@@ -770,14 +763,14 @@
 				3525449C1E663D2C004C8F1C /* PBXTargetDependency */,
 				35B711D61E5E7AD2001EDA8D /* PBXTargetDependency */,
 			);
-			name = MapboxNavigationUITests;
+			name = MapboxNavigationTests;
 			productName = MapboxNavigationUITests;
```
Comment (1ec5) [inline MapboxNavigation.xcodeproj/project.pbxproj]: Try changing this to MapboxNavigationTests in a text editor.

File: MapboxNavigation.xcodeproj/project.pbxproj [hunk 9]
```diff
@@ -788,14 +781,14 @@
 			);
 			dependencies = (
 			);
-			name = MapboxNavigation;
+			name = MapboxCoreNavigation;
 			productName = MapboxNavigation;
```
Comment (1ec5) [inline MapboxNavigation.xcodeproj/project.pbxproj]: Try changing this to MapboxCoreNavigation in a text editor.

File: MapboxNavigation.xcodeproj/project.pbxproj [hunk 10]
```diff
@@ -808,9 +801,9 @@
 				C5ADFBD51DDCC7840011824B /* PBXTargetDependency */,
 				352BBC3F1E5E6ADC00703DF1 /* PBXTargetDependency */,
 			);
-			name = MapboxNavigationTests;
+			name = MapboxCoreNavigationTests;
 			productName = MapboxNavigationTests;
```
Comment (1ec5) [inline MapboxNavigation.xcodeproj/project.pbxproj]: Try changing this to MapboxCoreNavigationTests in a text editor.

File: MapboxNavigationTests/CocoaPodsTest/PodInstall/Podfile.lock [hunk 11]
```diff
@@ -1,72 +0,0 @@
-PODS:
```
Comment (1ec5) [inline MapboxNavigationTests/CocoaPodsTest/PodInstall/Podfile.lock]: It looks like this file was removed instead of being moved under MapboxCoreNavigationTests.

**Comment by bsudekum** [comment]
@1ec5 @frederoni @ericrwolfe this is ready for review

File: Examples/Objective-C/Info.plist [hunk 12]
```diff
@@ -23,7 +23,7 @@
 	<key>LSRequiresIPhoneOS</key>
 	<true/>
 	<key>MGLMapboxAccessToken</key>
-	<string>pk.feedCafeDadeDeadBeef-BadeBede.FadeCafeDadeDeed-BadeBede</string>
+	<string>pk.eyJ1IjoiYm9iYnlzdWQiLCJhIjoiTi16MElIUSJ9.Clrqck--7WmHeqqvtFdYig</string>
```
Comment (1ec5) [inline Examples/Objective-C/Info.plist]: Cycle this token, now that it’s public. Then revert this change.

File: MapboxCoreNavigation.podspec [hunk 13]
```diff
@@ -2,12 +2,12 @@ Pod::Spec.new do |s|
 
   # ―――  Spec Metadata  ―――――――――――――――――――――――――――――――――――――――――――――――――――――――――― #
 
-  s.name = "MapboxNavigation.swift"
-  s.version = "0.1.0"
+  s.name = "MapboxCoreNavigation"
+  s.version = "0.2.0"
   s.summary = "Mapbox Navigation library"
```
Comment (1ec5) [inline MapboxCoreNavigation.podspec]: The summary and description for this podspec should differ from the summary and description in the MapboxNavigation podspec.

**Comment by bsudekum** [comment]
This ready to go

File: MapboxCoreNavigation.podspec [hunk 14]
```diff
@@ -2,12 +2,12 @@ Pod::Spec.new do |s|
 
   # ―――  Spec Metadata  ―――――――――――――――――――――――――――――――――――――――――――――――――――――――――― #
 
-  s.name = "MapboxNavigation.swift"
-  s.version = "0.1.0"
-  s.summary = "Mapbox Navigation library"
+  s.name = "MapboxCoreNavigation"
+  s.version = "0.2.0"
+  s.summary = "Mapbox Core Navigation SDK"
```
Comment (1ec5) [inline MapboxCoreNavigation.podspec]: > Core components for turn-by-turn navigation on iOS.

File: MapboxCoreNavigation.podspec [hunk 15]
```diff
@@ -2,12 +2,12 @@ Pod::Spec.new do |s|
...
 
   s.description  = <<-DESC
-  MapboxNavigation.swift makes it easy to get step by step instructions for guiding a user along a route.
+  The Mapbox Navigation SDK makes it easy to get step by step instructions for guiding a user along a route.
```
Comment (1ec5) [inline MapboxCoreNavigation.podspec]: > Mapbox Core Navigation provides the core spatial and timing logic for turn-by-turn navigation along a route. For a complete turn-by-turn navigation interface, use the Mapbox Navigation SDK for iOS (`MapboxNavigation`).

File: MapboxNavigation.podspec [hunk 16]
```diff
@@ -2,12 +2,12 @@ Pod::Spec.new do |s|
 
   # ―――  Spec Metadata  ―――――――――――――――――――――――――――――――――――――――――――――――――――――――――― #
 
-  s.name = "MapboxNavigationUI.swift"
-  s.version = "0.1.0"
-  s.summary = "Mapbox Navigation UI library"
+  s.name = "MapboxNavigation"
+  s.version = "0.2.0"
+  s.summary = "Mapbox Navigation SDK"
```
Comment (1ec5) [inline MapboxNavigation.podspec]: > Complete turn-by-turn navigation interface for iOS.

File: MapboxNavigation.podspec [hunk 17]
```diff
@@ -2,12 +2,12 @@ Pod::Spec.new do |s|
...
 
   s.description  = <<-DESC
-  MapboxNavigationUI.swift makes it easy to get step by step UI for guiding a user along a route.
+  The Mapbox Navigation SDK makes it easy to get step by step UI for guiding a user along a route.
```
Comment (1ec5) [inline MapboxNavigation.podspec]: > The Mapbox Navigation SDK for iOS is a drop-in interface for turn-by-turn navigation along a route, complete with a well-designed map and easy-to-understand spoken directions. Routes are powered by [Mapbox Directions](https://www.mapbox.com/directions/).

File: README.md [hunk 18]
```diff
@@ -1,8 +1,8 @@
 # Mapbox Navigation SDK for iOS
 
-![Mapbox Navigation UI](https://cloud.githubusercontent.com/assets/764476/23636459/567771d2-028a-11e7-95cf-a8832792c67a.png) ![](https://cloud.githubusercontent.com/assets/764476/23671279/883c63ae-031f-11e7-8396-b404d18881e1.png)
+![Mapbox Navigation SDK](https://cloud.githubusercontent.com/assets/764476/23636459/567771d2-028a-11e7-95cf-a8832792c67a.png) ![](https://cloud.githubusercontent.com/assets/764476/23671279/883c63ae-031f-11e7-8396-b404d18881e1.png)
```
Comment (1ec5) [inline README.md]: This readme also contains the line:

> Mapbox Navigation core components (`MapboxNavigation` module)

which needs to turn into:

> Mapbox Core Navigation (`MapboxCoreNavigation` module)
